### ISSUE TITLE
[Broken] Soundcloud Go+ sends back an HTTP 400 Error

### ISSUE THREAD

**Issue by htucker77** [yt-dlp/yt-dlp#265]
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running yt-dlp version **2021.04.22**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
[debug] Command-line config: ['-u', 'PRIVATE', '-p', 'PRIVATE', '-x', 'https://soundcloud.com/marcusschossow/zedd-feat-echosmith-illution-marcus-schossow-years-remix-premaster/s-8wR08', '--verbose']
[debug] Loading archive file None
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] yt-dlp version 2021.04.22 (zip)
[debug] Python version 3.9.4 (CPython 64bit) - macOS-10.15.7-x86_64-i386-64bit
[debug] exe versions: ffmpeg 4.4, ffprobe 4.4, rtmpdump 2.4
[debug] Proxy map: {}
[soundcloud] Downloading JSON metadata
ERROR: Unable to download JSON metadata: HTTP Error 400: Bad Request (caused by <HTTPError 400: 'Bad Request'>); please report this issue on https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 666, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 2890, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/local/Cellar/python@3.9/3.9.4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py", line 523, in open
    response = meth(req, response)
  File "/usr/local/Cellar/python@3.9/3.9.4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py", line 632, in http_response
    response = self.parent.error(
  File "/usr/local/Cellar/python@3.9/3.9.4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py", line 561, in error
    return self._call_chain(*args)
  File "/usr/local/Cellar/python@3.9/3.9.4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py", line 494, in _call_chain
    result = func(*args)
  File "/usr/local/Cellar/python@3.9/3.9.4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py", line 641, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)


```

## Description

I was trying to download a track from soundcloud using my SC Go plus account. It throws an HTTP 400 error though instead. This method used to work in the past so I'm wondering if something broke on Soundcloud's side or on here.

**Comment by pukkandan** [comment]
Try with cookies
https://github.com/ytdl-org/youtube-dl#how-do-i-pass-cookies-to-youtube-dl

**Comment by htucker77** [comment]
> Try with cookies
> https://github.com/ytdl-org/youtube-dl#how-do-i-pass-cookies-to-youtube-dl

It worked this time, but it only downloaded the 128kbps mp3

**Comment by pukkandan** [comment]
is higher bitrate available?
Show output of `yt-dlp --cookies <cookies> -vUF <URL>`

**Comment by htucker77** [comment]
[soundcloud] marcusschossow/zedd-feat-echosmith-illution-marcus-schossow-years-remix-premaster: Downloading info JSON
[soundcloud] 221876571: Downloading JSON metadata
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for 221876571:
ID          EXT RESOLUTION | PROTO  | ACODEC   ABR
----------- --- ---------- - ------ - ------- ----
hls_mp3_128 mp3 audio only | m3u8_n | unknown 128k

Nope. Maybe it's because the track is private?

**Comment by htucker77** [comment]
There wasn't a way to select a bitrate. Previously it just gave an M4A of the track if the original file was uploaded in high quality. I can send over my details via PM over discord. harrison#5344

**Comment by pukkandan** [comment]
Seems like this is indeed an issue with obtaining oauth: https://www.reddit.com/r/youtubedl/comments/mx1lka/download_high_quality_soundcloud_audio

**Comment by htucker77** [comment]
Alright so I just tried using the method you linked and it does work. So something must have changed in soundcloud's authentication process and broke the previous method. I think doing it this way will work for now, but most likely it'll have to be fixed in the future.

**Comment by pukkandan** [comment]
@htucker77 Since you aren't in yt-dlp discord server, I cant send PMs to you. So I have sent a friend request

**Comment by JunkiEDM** [comment]
Using your oauth token will almost always work (note that it can change from time to time), but the username/password/netrc authentication will throw the 400 error in yt-dlp, whereas in youtube-dl, it won't throw a 400 error and instead just download the 128kbps mp3. 
Either way, it seems like there was a change made on Soundcloud's side to trigger this, since it broke without updating yt-dlp

Soundcloud has been making frequent changes to their site's styling and structure under-the-hood (and some publicly, notably the track/repost-based "stories" section), and it seems likely a Soundcloud redesign is imminent, so I wouldn't be surprised if they decide to tweak authentication further.

**Comment by llacb47** [comment]
> it seems likely a Soundcloud redesign is imminent

Oh no.

**Comment by blackjack4494** [comment]
Indeed some things changed with soundcloud.
To actually get the playlist (m3u8) one need to get the transcoding links which is already happening. Then it's needed to request the actual link for the playlist - this is done by querying this endpoint
`https://api-v2.soundcloud.com/media/soundcloud:tracks:<trackID>/<transcodingIdentifier>/stream/hls`  
To send a proper request one would need an Authorization Header including the OAuth.  
There are multiple ways to get the OAuth however the simplest method is to assume you are logged in (either thru credentials or by provided cookies).  
The required endpoint
`https://api-auth.soundcloud.com/connect/session/token?client_id=<clid>`
Do note that the client_id is needed here but we got it anyway.  
The return looks like this
`{"session":{"access_token":"2-2xxx-2xxx-gTIxxxx"}}`
`access_token` equals `oauth_token`

So far I saw two endpoints where oauth is needed:  (1) when we need to call the tracks endpoint to retrieve json and (2) when we need to call the hls endpoint to retrieve the actual link to the playlist. This let me wonder what happens if you try to use an oauth of a soundcloud account without go+ subscription. Will we get some access forbidden or the playlist anyway? That would allow basically to get the go+ stuff without subscription. One could think of other possibilities that I shouldn't mention here.

In simple terms an easy fix.
OAuth is also used on several other endpoints.

**Comment by pukkandan** [comment]
closing since #997 and #469
